### PR TITLE
add code generation README

### PR DESCRIPTION
**Pull request by bashbaug**
This PR tidies up a few minor issues related to the code generation PR #161:

1. It adds a README for the header code generation, similar to the README for ICD loader dispatch table code generation.
2. It bumps the version requirement for Mako from 1.1.0 to 1.2.2.

### CODE REVIEW

File: scripts/README.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# OpenCL Header Code Generation
+
+## Introduction
+
+In order to ease maintenance and enable faster development in the OpenCL headers the OpenCL headers for extensions are generated from the OpenCL XML machine readable grammar.
```
Comment (EwanC) [inline scripts/README.md:5]: A comma here reads slightly better to me
```suggestion
In order to ease maintenance and enable faster development in the OpenCL headers, the OpenCL headers for extensions are generated from the OpenCL XML machine readable grammar.
```

**Review by EwanC — DISMISSED** [review]

**Comment by bashbaug** [comment]
@EwanC would you mind approving these changes again? Applying your comma fix apparently dismissed your review 😄

**Review by Kerilk — APPROVED** [review]
Looking good.

File: scripts/requirements.txt [hunk 2]
```diff
@@ -1 +1 @@
-Mako==1.1.0
+Mako>=1.2.2
```
[the remark this answers is not in the document]
Comment (Kerilk) [reply scripts/requirements.txt:1]: Ha yes, this is a very good reason to upgrade the required version indeed, better than disregarding the advisory. Thanks!

**Pull request merged**